fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -143,8 +149,7 @@ public CollectionData calculateLoanCollectionData(final Long loanId) {
                 collectionData.setLastRepaymentAmount(lastRepaymentTransaction.getAmount());
             }
 
-            enrichWithDelinquencyPausePeriodInfo(collectionData, retrieveLoanDelinquencyActions(loanId),
-                    ThreadLocalContextUtil.getBusinessDate());
+            enrichWithDelinquencyPausePeriodInfo(collectionData, savedDelinquencyList, ThreadLocalContextUtil.getBusinessDate());
```

Comment:
The effective delinquency list is already calculated in line 133, so we can add that instead of the full savedDelinquencyList, then the enrich method would not need to do this calculaiton.